server/core/db/boltdb.go

```diff
@@ -101,28 +109,90 @@ func (b *BoltDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock,
 			currLock = newLock
 			return nil
 		}
+		// checking if current lock is with the same PR
+		var currLock models.ProjectLock
```

Comment:
@lilincmu this block was previously in an "else" statement in our branch, and then the double unmarshalling made more sense, perhaps indeed this is a forgotten cleanup (we actually didn't need the "else" since the "if" returns early). We will clean it up.

As to whether `currLock` it meant to shadow the `currLockSerialized`:96 - not sure what you mean by "shadow", but is meant to be the same lock but deserialized, so that we can check if it refers to the same PR, in which case we just silently do nothing, since the lock is already there.

> If so, I think the outer currLock will be empty, and we will not be returning the current lock as expected when another PR is trying to acquire the lock, right?

That part I am not getting.. maybe you could elaborate.

Reply:
That's actually true, it was a bug and now it's fixed in  https://github.com/runatlantis/atlantis/pull/2782 with  https://github.com/runatlantis/atlantis/pull/2782/commits/6fab4921dcbcf9588e7030cf9f4e97c311f9bbc8#diff-9af0e8752843e8fe113c7531eb04cdbbb018432ce7fbbd31eba369b456cccfe3